src/hybrid.js

```diff
@@ -0,0 +1,325 @@
+import * as API from './api.js'
+import * as Node from './node.js'
+import * as ZeroComm from './zero-comm.js'
+import * as Proof from './proof.js'
+import { pow2 } from './uint64.js'
+
+/**
+ * We allow up to 2 ** 60 leafs in the tree, with is greater than then
+ * Number.MAX_SAFE_INTEGER ((2 ** 53) - 1) which is why we need to use
+ * uint64s.
+ */
+export const MAX_LOG2_LEAFS = 60
+
+/**
+ * @param {number} log2Leafs
+ * @returns {API.AggregateTree}
+ */
+export const create = (log2Leafs) => {
+  if (log2Leafs > MAX_LOG2_LEAFS) {
+    throw new RangeError(`too many leafs: 2 ** ${log2Leafs}`)
+  }
+
+  if (log2Leafs < 0) {
+    throw new RangeError(`cannot have negative log2Leafs`)
+  }
+
+  return new Hybrid(log2Leafs)
+}
+
+class Hybrid {
```

Comment:
I actually want to rename it to `AggregateTree` it's just go implementation called it `Hybrid` and I forgot I wanted to rename things.

Reply:
`AggregateTree` was actually first suggestion I wrote here. Ended up making Hybrid because it looked by the name that it would allow not only aggregates